### ISSUE TITLE
Import "Checkmarx Scan" failed when filename is empty

### ISSUE THREAD

**Issue by rucciva** [DefectDojo/django-DefectDojo#5636]
**Bug description**
I don't know if this is a bug in checkmarx or not but its report can contain node with empty filename. When uploaded to defect dojo, it will throw exception

**Steps to reproduce**
Steps to reproduce the behavior: Upload scans from checkmarx result than contains empty filename on some of the query

**Expected behavior**
No Error

**Deployment method** *(select with an `X`)*
- [x] Docker Compose
- [ ] Kubernetes
- [ ] GoDojo

**Environment information**
 - Operating System: Ubuntu 20.04
 - DefectDojo version (see footer) or commit message: 2.5.0

**Logs** 

```
uwsgi_1         | [24/Dec/2021 05:05:41] ERROR [dojo.engagement.views:625] expected str, bytes or os.PathLike object, not NoneType
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/./dojo/engagement/views.py", line 610, in import_scan_results
uwsgi_1         |     test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, user, environment, active=active, verified=verified, tags=tags,
uwsgi_1         |   File "/app/./dojo/importers/importer/importer.py", line 331, in import_scan
uwsgi_1         |     parsed_findings = parser.get_findings(scan, test)
uwsgi_1         |   File "/app/./dojo/tools/checkmarx/parser.py", line 97, in get_findings
uwsgi_1         |     self.process_result_file_name_aggregated(dupes, findingdetail, query, result, find_date)
uwsgi_1         |   File "/app/./dojo/tools/checkmarx/parser.py", line 113, in process_result_file_name_aggregated
uwsgi_1         |     title = "{} ({})".format(titleStart, ntpath.basename(sinkFilename))
uwsgi_1         |   File "/usr/local/lib/python3.8/ntpath.py", line 216, in basename
uwsgi_1         |     return split(p)[1]
uwsgi_1         |   File "/usr/local/lib/python3.8/ntpath.py", line 185, in split
uwsgi_1         |     p = os.fspath(p)
uwsgi_1         | TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Sample scan files (truncated)**
```
<?xml version="1.0" encoding="utf-8"?>
<CxXMLResults InitiatorName="support" Owner="Administrator" ScanId="1003406" ProjectId="10" ProjectName="project" TeamFullPathOnReportDate="CxServer" DeepLink="https://checkmarx/CxWebClient/ViewerMain.aspx?scanid=1003406&amp;projectid=10" ScanStart="Friday, December 24, 2021 9:12:14 AM" Preset="Checkmarx Default" ScanTime="00h:28m:02s" LinesOfCodeScanned="233670" FilesScanned="1063" ReportCreationTime="Friday, December 24, 2021 9:40:43 AM" Team="CxServer" CheckmarxVersion="9.4.0.2076" ScanComments="" ScanType="Full" SourceOrigin="LocalPath" Visibility="Public">
  <Query id="5437" cweId="346" name="Missing_HSTS_Header" group="PHP_Medium_Threat" Severity="Medium" Language="PHP" LanguageHash="0487062789656422" LanguageChangeDate="2021-08-24T00:00:00.0000000" SeverityIndex="2" QueryPath="PHP\Cx\PHP Medium Threat\Missing HSTS Header Version:1" QueryVersionCode="119183642">
    <Result NodeId="10034060073" FileName="" Status="Recurrent" Line="1" Column="0" FalsePositive="True" Severity="Medium" AssignToUser="" state="1" Remark="some remark, [Monday, October 25, 2021 9:31:40 AM]: Changed status to Not Exploitable" DeepLink="https://checkmarx/CxWebClient/ViewerMain.aspx?scanid=1003406&amp;projectid=10&amp;pathid=73" SeverityIndex="2" StatusIndex="1" DetectionDate="9/13/2021 5:12:37 AM">
      <Path ResultId="1003406" PathId="73" SimilarityId="445614419">
        <PathNode>
          <FileName></FileName>
          <Line>1</Line>
          <Column>0</Column>
          <NodeId>1</NodeId>
          <Name>array</Name>
          <Type></Type>
          <Length>0</Length>
        </PathNode>
      </Path>
    </Result>
  </Query>
</CxXMLResults>
```

**Screenshots**

**Additional context** (optional)

**Comment by damiencarol** [comment]
Taking this one.

**Comment by damiencarol** [comment]
Just out of curiosity, how is it possible to have an empty file name?

**Comment by rucciva** [comment]
<img width="1440" alt="Screen Shot 2021-12-24 at 17 08 55" src="https://user-images.githubusercontent.com/4647473/147343540-8d01d6c7-68e4-44dc-bec3-2665ea50d542.png">

I think because it was marked as Not Exploitable but it no longer appear in recent scans

**Comment by damiencarol** [comment]
@rucciva pushed a first patch

**Comment by rucciva** [comment]
Thanks @damiencarol . Is there any docker image that i can use to test the patch?

**Comment by damiencarol** [comment]
I don't think so. you can build it by getting my branch and building it yourself.

**Comment by rucciva** [comment]
Hi @damiencarol , now it produces this error
```
uwsgi_1         | [28/Dec/2021 08:57:19] ERROR [dojo.engagement.views:625] add_language() got an unexpected keyword argument 'files'
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/./dojo/engagement/views.py", line 610, in import_scan_results
uwsgi_1         |     test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, user, environment, active=active, verified=verified, tags=tags,
uwsgi_1         |   File "/app/./dojo/importers/importer/importer.py", line 331, in import_scan
uwsgi_1         |     parsed_findings = parser.get_findings(scan, test)
uwsgi_1         |   File "/app/./dojo/tools/checkmarx/parser.py", line 96, in get_findings
uwsgi_1         |     add_language(test.engagement.product, lang, files=language_list[lang])
uwsgi_1         | TypeError: add_language() got an unexpected keyword argument 'files'
```

**Comment by damiencarol** [comment]
I missed a file in the PR :sweat: . It should be good now.

**Comment by rucciva** [comment]
it works now, thanks.

**Comment by StefanFl** [comment]
merged for release 2.6.0
